packages/safe-core-sdk/src/managers/contractManager.ts

```diff
@@ -0,0 +1,72 @@
+import { Provider } from '@ethersproject/providers'
+import { Signer } from 'ethers'
+import { GnosisSafe, MultiSend } from '../../typechain'
+import MultiSendAbi from '../abis/MultiSendAbi.json'
+import SafeAbiV120 from '../abis/SafeAbiV1-2-0.json'
+import { ContractNetworksConfig, defaultContractNetworks } from '../configuration/contracts'
+
+class ContractManager {
+  #safeContract!: GnosisSafe
+  #multiSendContract!: MultiSend
+
+  static async create(
+    ethers: any,
+    safeAddress: string,
+    chainId: number,
+    currentProviderOrSigner: Provider | Signer,
+    provider: Provider,
+    contractNetworks?: ContractNetworksConfig
+  ) {
+    const safeSdk = new ContractManager()
+    await safeSdk.init(
+      ethers,
+      safeAddress,
+      chainId,
+      currentProviderOrSigner,
+      provider,
+      contractNetworks
+    )
+    return safeSdk
+  }
+
+  async init(
+    ethers: any,
+    safeAddress: string,
+    chainId: number,
+    currentProviderOrSigner: Provider | Signer,
+    provider: Provider,
+    contractNetworks?: ContractNetworksConfig
+  ) {
+    const contractNetworksConfig = { ...defaultContractNetworks, ...contractNetworks }
+    const contracts = contractNetworksConfig[chainId]
+    if (!contracts) {
+      throw new Error('Safe contracts not found in the current network')
+    }
+
+    const safeContractCode = await provider.getCode(safeAddress)
+    if (safeContractCode === '0x') {
+      throw new Error('Safe Proxy contract is not deployed in the current network')
+    }
+    this.#safeContract = new ethers.Contract(safeAddress, SafeAbiV120, currentProviderOrSigner)
+
+    const multiSendContractCode = await provider.getCode(contracts.multiSendAddress)
+    if (multiSendContractCode === '0x') {
+      throw new Error('MultiSend contract is not deployed in the current network')
+    }
+    this.#multiSendContract = new ethers.Contract(
+      contracts.multiSendAddress,
+      MultiSendAbi,
+      currentProviderOrSigner
+    )
+  }
+
+  get safeContract() {
```

Comment:
i'm not an oop expert but this getter is the same as just accessing the property, why make it private?

Reply:
to prevent from being modified

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields#private_instance_fields

TIL you also can't access them otherwise. Btw, in safe-apps-sdk we don't use it anymore because of https://github.com/gnosis/safe-apps-sdk/issues/141